### PR TITLE
Add blocking in UserDataManager to make queue of requests

### PR DESCRIPTION
**Pull request by irina060981**
1) Add `blocked` property to UserDataManager to handle the state, when DBAdapter is busy on requests
2) Add `requestsQueue` property to UserDataManager to store a queue of requests
3) Add mechanism to put requests to the queue when UserDataManager is already blocked and execute all requests (by order) fro the queue when UserDataManager becomes unblocked

For the following issue:
https://github.com/alpheios-project/wordlist/issues/4

### CODE REVIEW

**Comment by irina060981** [comment]
@balmas allowed to merge without a review.

File: src/controllers/user-data-manager.js [hunk 1]
```diff
@@ -172,6 +206,13 @@ export default class UserDataManager {
     return [...remoteDataItems,...addToRemote]
   }
 
+  checkRequestQueue () {
+    if (this.requestsQueue.length > 0) {
```
Comment (balmas) [inline src/controllers/user-data-manager.js:210]: @irina060981 should this be in a while loop in case there is more than one request queued?

Comment (irina060981) [reply src/controllers/user-data-manager.js:210]: No it shouldn't. The workflow here is the following:
At the satrt of each modifying method (update, deleteOne, deleteMany) it checks if some modifing method is currently in use (by this.blocking property).
If it is blocked, than it push a method execution parameters to the queue.
When the method finished to block (at the end of the execution), it executes checkRequestQueue. 
This method takes the first method from the queue and executes (and inside it the same blocking workflow is made).
This way checkRequestQueue will be executed each time the modifying method ends until queue becomes empty.

Comment (balmas) [reply src/controllers/user-data-manager.js:210]: ok, thanks for the clarification!

**Pull request merged**